source/PlayerInfo.cpp

```diff
@@ -743,7 +743,7 @@ const shared_ptr<Ship> &PlayerInfo::FlagshipPtr()
 	if(!flagship)
 	{
 		for(const shared_ptr<Ship> &it : ships)
-			if(!it->IsParked() && it->GetSystem() == system && it->CanBeFlagship())
+			if(!it->IsParked() && it->GetSystem() == system && planet->CanLand(*it) && it->CanBeFlagship())
```

Comment:
There's no guarantee the player is landed when this function is called, which means we would then dereference a null pointer.

Reply:
I think that in practice flagship is null only when you are landed, since the flagship appears to only be reset when you land on a planet and do stuff with ships (reorder, sell, disown, etc.).
What do you think about changing line 743 to `if(!flagship && planet)` ?